pipeline-model-extensions/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/model/BuildCondition.java

```diff
@@ -45,13 +46,45 @@
  */
 public abstract class BuildCondition implements Serializable, ExtensionPoint {
 
-    public abstract boolean meetsCondition(WorkflowRun r);
+    /**
+     * @deprecated since 1.2
+     */
+    @Deprecated
+    public boolean meetsCondition(WorkflowRun r) {
+        if (Util.isOverridden(BuildCondition.class, getClass(), "meetsCondition", WorkflowRun.class)) {
+            return getClass().cast(this).meetsCondition(r);
+        } else {
+            return true;
+        }
+    }
+
+    public boolean meetsCondition(WorkflowRun r, Exception error) {
+        if (Util.isOverridden(BuildCondition.class, getClass(), "meetsCondition", WorkflowRun.class, Exception.class)) {
```

Comment:
This ... looks ... a bit strange.
If the method is overridden this code wouldn't run anyway?